storage/engine/mvcc.go

```diff
@@ -609,61 +586,60 @@ func mvccGetInternal(engine Engine, key roachpb.Key, metaKey MVCCKey,
 
 		// We want to know if anything has been written ahead of timestamp, but
 		// before MaxTimestamp.
-		nextKey := MVCCEncodeVersionKey(key, txn.MaxTimestamp)
-		var err error
-		valueKey, err = getValue(engine, nextKey, MVCCEncodeKey(key.Next()), value)
-		if err != nil {
-			return nil, nil, err
-		}
-		if valueKey != nil {
-			_, ts, _, err := MVCCDecodeKey(valueKey)
-			if err != nil {
-				return nil, nil, err
-			}
-			if timestamp.Less(ts) {
-				// Third case: Our read timestamp is sufficiently behind the newest
-				// value, but there is another previous write with the same issues
-				// as in the second case, so the reader will have to come again
-				// with a higher read timestamp.
-				return nil, nil, &roachpb.ReadWithinUncertaintyIntervalError{
-					Timestamp:         timestamp,
-					ExistingTimestamp: ts,
-				}
-			}
-		}
-		// Fourth case: There's no value in our future up to MaxTimestamp, and
-		// those are the only ones that we're not certain about. The correct
-		// key has already been read above, so there's nothing left to do.
+		seekKey = MVCCEncodeVersionKey(key, txn.MaxTimestamp)
+		checkValueTimestamp = true
 	} else {
 		// Fifth case: We're reading a historic value either outside of
```

Comment:
I can't see those steps since it's all one commit, but it looks alright to me and as long as the tests pass I'm comfortable with it.